wasm/src/lib.rs

```diff
@@ -78,7 +79,20 @@ pub fn get_qir(
 ) -> Result<String, String> {
     let language_features = LanguageFeatures::from_iter(language_features);
     let sources = get_source_map(sources, &None);
-    _get_qir(sources, language_features)
+    if language_features.contains(LanguageFeatures::PreviewQirGen) {
+        // the use of AdaptiveProfileQirGen is currently being used to
+        // determine what profile to target as the get_qir interface doesn't
+        // have a way to specify the profile to target and the old codegen
+        // is hardcoded to use the Base profile.
```

Comment:
I'm not a fan of having logic in the WASM layer. Can we just plumb `TargetProfile` through from TypeScript (`compiler.ts`)? 

`targetProfile` can be added to `ProgramConfig` as an optional property, I think that's perfectly appropriate and backward compatible (I can take care of updating `getCircuit` to be consistent later).

https://github.com/microsoft/qsharp/blob/e3fa6744610bc253365f0c8372e27721d0e0f8ac/npm/qsharp/src/compiler/compiler.ts#L92-L97